docs/pages/reference/helm-reference/teleport-cluster.mdx

```diff
@@ -1892,6 +1892,25 @@ Boolean value that specifies whether to generate a Kubernetes `Ingress` for the
     enabled: true
   ```
 
+## `ingress.useExisting`
+
+| Type      | Default value | Required? |
+|-----------|---------------|-----------|
+| `boolean` | `false`       | No        |
+
+
+`ingress.useExisting` indicates to the chart that you are providing your own ingress.
+The chart will configure Teleport like it's running behind an ingress, but will not
+create the ingress resource. You are responsible for creating and managing the ingress.
+
+`values.yaml` example:
+
+```yaml
+ingress:
```

Comment:
should we start renaming this to the gateway api now that ingress api is frozen for quite some time?

Reply:
Last time I checked the gateway API is optional, its CRDs are not here by default. This makes the adoption for generic tools like us very hard.

I can extend the `ingress.useExisting` documentation that users can manage their own ingress, or HTTProute if they want. Using TCPRoute is roughly equivalent to using a ServiceLB and we must not allow self-signed+replication in this setup. TLSRoute experience will depend on the configuration. I don't think we can just replace "ingress" by "gateway API" as the new gwapi is way more powerful and allows user to shoot themselves in the foot in brand new ways.

Users are also thinking about ingress when they need to expose a service (and they also sadly don't consider L4 LoadBalancer Services, this just makes Teleport slower and more complex to deploy). I think ingress is the name everyone is looking for and we should not rename this section.